File: backend/src/main/java/com/documed/backend/referrals/Referral.java
package com.documed.backend.referrals;

import com.documed.backend.visits.model.Visit;
import java.util.Date;
import lombok.Data;
import lombok.NonNull;

@Data
public class Referral {
  private int id;
  private String diagnosis;
  @NonNull private ReferralType type;
  private final Visit visit;
  private Date expirationDate;
}

File: frontend/src/shared/types/Referral.ts
import { ReferralType } from './enums';
import { VisitLite } from './Visit';

export interface Referall extends ReferralLite {
  visit: VisitLite;
}

export interface ReferralLite {
  id: number;
  diagnosis: string | null;
  type: ReferralType;
  expirationDate: Date | null;
}
